cloud_governance/common/clouds/aws/iam/iam_operations.py

```diff
@@ -158,3 +159,141 @@ def untag_role(self, role_name: str, tags: list):
+                        else:
+                            last_used_days = "N/A"
+                    except Exception:
+                        last_used_days = "N/A"
+
+                    result[username][label] = [status, age_days, last_used_days]
+
+                # Most recent access key activity
+                result[username]["last_activity_days"] = min(last_used_days_list) if last_used_days_list else "N/A"
+
+                # Tags as list of dicts
+                try:
+                    tag_response = self.iam_client.list_user_tags(UserName=username)
+                    tags = tag_response.get('Tags', [])
+                except Exception:
+                    tags = []
+
+                result[username]["tags"] = tags
+                result[username]["region"] = region_name
+                result[username]["ResourceId"] = user.get('UserId')  # <-- Unique ID
+        return result
+
+    def has_active_access_keys(self, username):
+        """
+        Checks if the given IAM user has any active access keys.
+        When no user access key return False
+        Args:
+            username (str): IAM user name
+
+        Returns:
+            bool: True if any access key is active, False otherwise
+        """
+        try:
+            access_keys = self.iam_client.list_access_keys(UserName=username)['AccessKeyMetadata']
+        except Exception as e:
+            logger.error(f"Failed to list access keys for user '{username}': {e}")
+            return False
+
+        for key in access_keys:
+            if key.get('Status') == 'Active':
+                return True
+        return False
+
+    def deactivate_user_access_keys(self, username):
+        """
+        Deactivates all active access keys for the given IAM user.
+
+        Args:
+            username (str): IAM user name
+        """
+        try:
+            access_keys = self.iam_client.list_access_keys(UserName=username)['AccessKeyMetadata']
+        except Exception as e:
+            logger.error(f"Failed to list access keys for user '{username}': {e}")
+            return
+
+        for idx, key in enumerate(access_keys, start=1):
+            label = f"Access key {idx}"
+            current_status = key['Status'].lower()
+            access_key_id = key['AccessKeyId']
+
+            if current_status == 'active':
+                try:
+                    self.iam_client.update_access_key(
+                        UserName=username,
+                        AccessKeyId=access_key_id,
+                        Status='Inactive'
+                    )
+                    logger.info(f"{label} deactivated for user '{username}'")
+                except Exception as e:
+                    logger.error(f"Failed to deactivate {label} for user '{username}': {e}")
+            else:
+                logger.info(f"{label} is already inactive for user '{username}'")
+
+        logger.info(f"Deactivate access keys for user '{username}' have been processed.")
```

Comment:
We need to deactivate unused keys, not all keys. Her we should pass the inactive key_id to deactivate.

Reply:
We deactivate it by resource_id=user_name.
You can find [here](https://github.com/redhat-performance/cloud-governance/blob/f38118882e88834c3e38307f73915b791b5104cf/cloud_governance/policy/helpers/aws/aws_policy_operations.py#L61-L62) the way we deactivate only specific user as other policies.
Pls let me know if it is ok.

Reply:
Yeah, but if the user having multiple access_keys and one is recently used and one is used 90days back. We should delete the key that not using.